src/firefly/js/drawingLayers/Catalog.js

```diff
@@ -180,3 +209,112 @@ function computeHighlightLayer(tableData, highlightedRow, columns) {
     return [obj,obj2];
 }
 
+function computeSelectedIdxAry(selectInfo) {
+    const si= SelectInfo.newInstance(selectInfo);
+    if (!si.getSelectedCount()) return null;
+    return (idx) => si.isSelected(idx);
+}
+
+
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+//--- The following are utility functions that that are called
+//--- by the toolbar buttons to filter, select, unfilter, and unselect
+//--- The are not used directly by the Catalog drawing layer but
+//--- are aware of the data structure
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+
+
+export function selectCatalog(pv,dlAry) {
+    var p= primePlot(pv);
+    var sel= p.attributes[PlotAttribute.SELECTION];
+    if (!sel) return;
+    const catDlAry= getLayers(pv,dlAry);
+    if (catDlAry.length) {
+        const tooBig= catDlAry.some( (dl) => dl.dataTooBigForSelection);
+        if (tooBig) {
+            showInfoPopup('Your data set is too large to select. You must filter it down first.',
```

Comment:
I thought about that but I am afraid it would be more confusing.  The user might wonder why the select does not show up sometimes.